models/surveys.go

```diff
@@ -14,13 +36,88 @@ type Survey struct {
-func AllSurveys() ([]*SurveySummary, error) {
-	rows, err := db.Query("SELECT id, shortname FROM survey.survey ORDER BY shortname ASC")
+type API struct {
+	AllSurveysStmt                    *sql.Stmt
+	GetSurveyStmt                     *sql.Stmt
+	GetSurveyByShortNameStmt          *sql.Stmt
+	GetSurveyByReferenceStmt          *sql.Stmt
+	GetSurveyIDStmt                   *sql.Stmt
+	GetClassifierTypeSelectorStmt     *sql.Stmt
+	GetClassifierTypeSelectorByIdStmt *sql.Stmt
+}
+
+func NewAPI(db *sql.DB) (*API, error) {
+	allSurveysSQL := "SELECT id, shortname FROM survey.survey ORDER BY shortname ASC"
+	allSurveyStmt, err := createStmt(allSurveysSQL, db)
+	if err != nil {
+		return nil, err
+	}
+
+	getSurveySQL := "SELECT id, shortname, longname, surveyref from survey.survey WHERE id = $1"
+	getSurveyStmt, err := createStmt(getSurveySQL, db)
+	if err != nil {
+		return nil, err
+	}
+
+	getSurveyByShortNameSQL := "SELECT id, shortname, longname, surveyref from survey.survey WHERE LOWER(shortName) = LOWER($1)"
+	getSurveyByShortNameStmt, err := createStmt(getSurveyByShortNameSQL, db)
+	if err != nil {
+		return nil, err
+	}
+
+	getSurveyByReferenceSQL := "SELECT id, shortname, longname, surveyref from survey.survey WHERE LOWER(surveyref) = LOWER($1)"
+	getSurveyByReferenceStmt, err := createStmt(getSurveyByReferenceSQL, db)
+	if err != nil {
+		return nil, err
+	}
+
+	getSurveyIdSQL := "SELECT id FROM survey.survey WHERE id = $1"
+	getSurveyIdStmt, err := createStmt(getSurveyIdSQL, db)
+	if err != nil {
+		return nil, err
+	}
+
+	getClassifierTypeSelectorSql := "SELECT classifiertypeselector.id, classifiertypeselector FROM survey.classifiertypeselector INNER JOIN survey.survey ON classifiertypeselector.surveyfk = survey.surveypk WHERE survey.id = $1 ORDER BY classifiertypeselector ASC"
+	getClassifierTypeSelectorStmt, err := createStmt(getClassifierTypeSelectorSql, db)
 	if err != nil {
 		return nil, err
 	}
 
+	getClassifierTypeSelectoreByIdSql := "SELECT id, classifiertypeselector, classifiertype FROM survey.classifiertype INNER JOIN survey.classifiertypeselector ON classifiertype.classifiertypeselectorfk = classifiertypeselector.classifiertypeselectorpk WHERE classifiertypeselector.id = $1 ORDER BY classifiertype ASC"
+	getClassifierTypeSelectorByIdStmt, err := createStmt(getClassifierTypeSelectoreByIdSql, db)
+	if err != nil {
+		return nil, err
+	}
+
+	return &API{
+		AllSurveysStmt:                    allSurveyStmt,
+		GetSurveyStmt:                     getSurveyStmt,
+		GetSurveyByShortNameStmt:          getSurveyByShortNameStmt,
+		GetSurveyByReferenceStmt:          getSurveyByReferenceStmt,
+		GetSurveyIDStmt:                   getSurveyIdStmt,
+		GetClassifierTypeSelectorStmt:     getClassifierTypeSelectorStmt,
+		GetClassifierTypeSelectorByIdStmt: getClassifierTypeSelectorByIdStmt,
+	}, nil
+}
+
+func (api *API) Info(w http.ResponseWriter, r *http.Request) {
```

Comment:
Ideally use a `/healthcheck` endpoint instead of `/info`

Reply:
info endpoint has a different function to what ah healthcheck endpoint does - its worth adding in a healthcheck endpoint tho, will raise and issue for this